lib/analyser.py

```diff
@@ -3,3 +3,134 @@
+        net_present_value += discounted_profit
+
+    terminal_value = profit*(1+perpetual_growth_rate)/(discount_rate-perpetual_growth_rate)
+    discounted_terminal_value = terminal_value/(pow((1+discount_rate),(year+1)))
+    net_present_value += discounted_terminal_value
+    return (market_cap/net_present_value)
+
+def DCF_MC_analysis(self, init_profit, init_fixed_asset, market_cap):
+    """
+    Usage:
+    Margin of safety parameter
+    Provides the (mean+3*sigma) limit of the market_cap/net_present_value estimates.
+    This limit < 0.9 for enough margin of safety.
+
+    Params:
+    1. init_profit = {CFO, (OP - Tax paid) (this is better)}
+    """
+    # TODO : run tests on iterations to see what value is good enough
+    iterations              = 100000
+    init_profit_growth_rate = np.random.normal(loc=0.15, scale=0.01, size=iterations)
+    perpetual_growth_rate   = [0.05, 0.06]
+    years_until_decay       = [5, 6, 7, 8]
+    discount_rate           = np.random.normal(loc=0.12, scale=0.01, size=iterations)
+    no_of_years             = 10
+    npv_distribution        = []
+    # TODO : Try to add a dataframe based operation here to check if it is faster
+    for yud in years_until_decay:
+        for pgr in perpetual_growth_rate:
+            for j in range(iterations):
+                npv_distribution.append(self.calc_NPV(init_profit, init_profit_growth_rate[j], \
+                pgr, yud, discount_rate[j], init_fixed_asset, \
+                market_cap, no_of_years)) 
+    return np.mean(npv_distribution)+3*np.std(npv_distribution)
+
+def Calc_SSGR_revenue(revenue, NPM, DPR, fixed_assets_start, depreciation):
+    """
+    Usage:
+    1. Margin of safety parameter
+    2. Calculate the self sustainable growth rate (SSGR) of the company on the basis of sales
+    3. SSGR indicates the company's ability to fund the extent of its growth without needing external debt
+    4. If SSGR > past sales growth trends then company doesn't need debt.
+    
+    Params:
+    1. revenue              : revenue of current year
+    2. NPM                  : Net profit margin
+    3. DPR                  : Dividend payout ratio
+    4. fixed_assets_start   : Net fixed assets at end of current year
+    5. depreciation         : Depreciation of the current year
+    """ 
+    # Funds available for reinvestment for investment from current year
+    funds_available = revenue*NPM*(1-DPR)
+    # Fixed assets at the beginning of the next year assuming all the funds available are reinvested
+    # Assumed deprec
+    fixed_assets_end = (fixed_assets_start*(1 - (depreciation/fixed_assets_start))) + funds_available
+    # Expected sales
+    expected_sales = fixed_assets_end*(revenue/fixed_assets_start)
+    SSGR = (expected_sales/revenue) - 1
+    return SSGR
+    #return (((((fixed_assets_start*(1 - (depreciation/fixed_assets_start))) + (revenue*NPM*(1-DPR)))*NFAT)/revenue) - 1)
+
+def Calc_SSGR_CFO(CFO, PAT, DPR, fixed_assets_start, depreciation):
```

Comment:
```suggestion
def calc_ssgr_cfo(cfo, pat, dpr, fixed_assets_start, depreciation):
```

Reply:
Followed the standard convention and implemented the changes